### DISCUSSION TITLE
Error in git push github actions

### DISCUSSION THREAD

**Discussion by Quantum-Codes** [community/community#25710] [Actions]
<p><a href="https://github.com/Quantum-Codes/Scratch-follow-graph" class="inline-onebox" rel="noopener nofollow ugc">GitHub - Quantum-Codes/Scratch-follow-graph: Python code that is run by github actions which generates graphs</a> — repo link.</p>
<p>I was using github actions to run my code to generate graphs and push them. It works inconsistently. Sometimes it works, sometimes it throws errors</p>
<p>Here is the error:</p>
<pre><code class="lang-auto">Run git config user.name github-actions 

 git config user.name github-actions 

 git config user.email github-actions@github.com 

 git add . 

 git commit -m "generated" 

 git push 

 shell: /usr/bin/bash -e {0} 

 env: 

 pythonLocation: /opt/hostedtoolcache/Python/3.10.2/x64 

 LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.2/x64/lib 

[main 13df927] generated 

 2 files changed, 63 insertions(+), 63 deletions(-) 

To https://github.com/Quantum-Codes/Scratch-follow-graph 

 ! [rejected] main -&gt; main (fetch first) 

error: failed to push some refs to 'https://github.com/Quantum-Codes/Scratch-follow-graph' 

hint: Updates were rejected because the remote contains work that you do 

hint: not have locally. This is usually caused by another repository pushing 

hint: to the same ref. You may want to first integrate the remote changes 

hint: (e.g., 'git pull ...') before pushing again. 

hint: See the 'Note about fast-forwards' in 'git push --help' for details. 

Error: Process completed with exit code 1.
</code></pre>
<p>Please dont give answers with force push <code>-f</code></p>
<ol>
<li>How to fix it?</li>
<li>Making a PR and merging manually will fix it?</li>
</ol>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Quantum-Codes" data-post="1" data-topic="226539">
<div class="title">
<div class="quote-controls"></div>
 Quantum-Codes:</div>
<blockquote>
<pre><code class="lang-nohighlight">error: failed to push some refs to 'https://github.com/Quantum-Codes/Scratch-follow-graph' 
hint: Updates were rejected because the remote contains work that you do 
hint: not have locally. This is usually caused by another repository pushing 
hint: to the same ref. You may want to first integrate the remote changes 
hint: (e.g., 'git pull ...') before pushing again. 
</code></pre>
</blockquote>
</aside>
<p>This looks like someone (e.g. you) or something (e.g. another Actions run) else pushed in between the workflow starting and pushing its result, so a fast-forward of the remote branch is impossible.</p>
<p>You’ll have to consider how you want your workflow to handle that situation. Making a PR instead of pushing directly would avoid the problem conflict, but you might end up with a lot of (possibly conflicting) PRs.</p>
